src/t8_forest/t8_forest.c

```diff
@@ -1292,14 +1292,72 @@ t8_forest_compute_elements_offset (t8_forest_t forest)
   T8_ASSERT (current_offset == forest->local_num_elements);
 }
 
-void
-t8_forest_write_vtk (t8_forest_t forest, const char *filename)
+int
+t8_forest_write_vtk_ext (t8_forest_t forest,
+                         const char *fileprefix,
+                         int write_treeid,
+                         int write_mpirank,
+                         int write_level,
+                         int write_element_id,
+                         int write_ghosts,
+                         int curved_flag,
+                         int do_not_use_API,
+                         int num_data, t8_vtk_data_field_t *data)
 {
   T8_ASSERT (forest != NULL);
   T8_ASSERT (forest->rc.refcount > 0);
   T8_ASSERT (forest->committed);
 
-  t8_forest_vtk_write_file (forest, filename, 1, 1, 1, 1, 1, 0, NULL);
+#if T8_WITH_VTK
+  if (!do_not_use_API) {
+    if (write_ghosts) {
+      t8_errorf ("Export of ghosts not yet available with the vtk API. "
+                 "Please use the inbuild function instead.\n"
+                 "Did not write any vtk output.\n");
+      return 0;
```

Comment:
We should not return 0 here, since writing was not successful.

Reply:
Until now, the vtk functions had different return values for exiting successfull or not. Now, all vtk functions return true if successfull and false of not. But we can change that the other way around. Is there already consistent way the other, not vtk related, functions handle this?

Reply:
Sorry, i mixed it up. 

!= 0 on success
0 on failure

is perfect.
My comment is wrong since you correctly return 0 here.